pypsa/optimization/abstract.py

```diff
@@ -694,6 +821,92 @@ def optimize_with_rolling_horizon(
                 )
         return self._n
 
+    def build_linexpr_from_weights(
```

Comment:
If the user will never want to use this (?), this should be internal
```suggestion
    def _build_linexpr_from_weights(
```